src/clojure/nrepl/misc.clj

```diff
@@ -99,14 +99,24 @@
   [:ns :name :doc :file :arglists :forms :macro :special-form
    :protocol :line :column :added :deprecated :resource])
 
-(defn handle-file-meta [file]
+(defn handle-file-meta-1 [file]
   (cond
     (nil? file) file
     (instance? URL file) (str file)
     (instance? java.io.File file) (str file)
     (not (nil? (io/resource file))) (str (io/resource file))
     :else file))
 
+(defn handle-file-meta [file]
+  (let [type-map {java.net.URL #(str %)
```

Comment:
My point was that you don't really need to apply `str` explicitly for each case, but rather you can just apply `str` once to whatever the `result` from `cond`. You can leave the `some` outside of the function if you want or put the nil handling outside the `cond` logic, so it wouldn't be converted to an emptry string.

Reply:
Hmm, last time I made a function with same logic, but for some reason the local tests did not pass, it seems I forgot to save the file and tests ran on the wrong version of code, sorry for constantly distracting you with my PR :(